core/src/main/java/cucumber/api/PickleTestStep.java

```diff
@@ -0,0 +1,65 @@
+package cucumber.api;
+
+import java.util.List;
+
+/**
+ * A test step matches a line in a Gherkin scenario or background.
+ */
+public interface PickleTestStep extends TestStep {
```

Comment:
I'm not sure about `PickleTestStep`. Is it a test step of a pickle? No pickles does not contain test steps but pickle steps. Is it a test step created from a pickle? No it is a test case that is created from a pickle. I'm considering `PickleStepTestStep`, maybe a bit cumbersome but clear on that it is a test steps created from a pickle step.

Reply:
I've renamed it to `PickleStepTestStep`. I'm generally not enamored by the `ClassNameParentClassName` pattern for the exact reasons you mentioned but anything else leads to clashes with the Gherkin namespace.